### PR TITLE
feat: Added delete and archive functionality

### PR DESCRIPTION
**Pull request by Hardiegogo**
Added delete and archive functionality:
- created archive and trash pages
- added trashNotes and archivedNotes in useNotes context
- updated NotesReducer
- added css for desktop version

### CODE REVIEW

File: src/reducers/notesReducer.jsx [hunk 1]
```diff
@@ -1,18 +1,56 @@
...
+    case "SET_NOTES":
+      return {
+        ...state,
+        notes: action.payload,
+      };
+    case "ADD_NOTE":
+      return {
+        ...state,
+        notes: action.payload,
+      };
```
Comment (AmishMishra11) [inline src/reducers/notesReducer.jsx]: here both dispatch functions do the same task so you can just delete one.

File: src/reducers/notesReducer.jsx [hunk 2]
```diff
@@ -1,18 +1,56 @@
...
+        ...state,
+        archivedNotes: action.archivedNotes,
+        notes: action.notes,
+      };
+    case "RESTORE_FROM_ARCHIVE":
+      return {
+        ...state,
+        archivedNotes: action.archivedNotes,
+        notes: action.notes,
+      };
```
Comment (AmishMishra11) [inline src/reducers/notesReducer.jsx]: same as above

**Review by AmishMishra11 — APPROVED** [review]
Clean code just make one small suggestion that I requested

File: src/components/notecard-component/NoteCard.css [hunk 3]
```diff
@@ -18,4 +18,26 @@
     font-size: 2.4rem;
     color:var(--coffee);
     margin-left: 8px;
+    cursor: pointer;
+}
+
+.active-icon{
+    color: var(--dark-coffee);
+    font-weight: 700;
```
Comment (stutisk) [inline src/components/notecard-component/NoteCard.css:26]: can use variables for font weight

**Review by stutisk — APPROVED** [review]
just a small suggestion , other then that everything looks good

**Pull request merged**